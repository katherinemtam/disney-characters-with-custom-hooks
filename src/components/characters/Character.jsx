import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, films, tvShows }) => {
  const stringyFilms = JSON.stringify(films);
  const stringyTvShows = JSON.stringify(tvShows);

  return (
    <>
      <h2>{name}</h2>
      <p>Films: {stringyFilms}</p>
      <p>TV Shows: {stringyTvShows}</p>
    </>
  );
};

Character.propTypes = {
  name: PropTypes.string.isRequired,
  films: PropTypes.arrayOf(PropTypes.string).isRequired,
  tvShows: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Character;
