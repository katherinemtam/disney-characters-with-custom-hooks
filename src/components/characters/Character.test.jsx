import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  const character = {
    name: 'Queen Arianna',
    films: ['Tangled', 'Tangled: Before Ever After'],
    tvShows: ['Once Upon a Time', 'Tangled: The Series'],
  };

  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(<Character {...character} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
