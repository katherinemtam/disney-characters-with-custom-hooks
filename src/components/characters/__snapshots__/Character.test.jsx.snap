// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <h2>
    Queen Arianna
  </h2>
  <p>
    Films: ["Tangled","Tangled: Before Ever After"]
  </p>
  <p>
    TV Shows: ["Once Upon a Time","Tangled: The Series"]
  </p>
</DocumentFragment>
`;
